packages/components/src/styles/components/app-header.scss

```diff
@@ -76,6 +75,30 @@
   .hds-dropdown-toggle-button,
   .hds-dropdown-toggle-icon {
     @include hds-interactive-dark-theme();
+
+    // disabled state:
+    &:disabled,
+    &[disabled],
+    &.mock-disabled,
+    &:disabled:focus,
+    &[disabled]:focus,
+    &.mock-disabled:focus,
+    &:disabled:hover,
+    &[disabled]:hover,
+    &.mock-disabled:hover {
+      @include hds-interactive-dark-theme-state-disabled();
+    }
+  }
+}
+
+// prevent interaction with AppHeader when SideNav is open/expanded
```

Comment:
For the record: I still think this is not the right solution, but the majority of the people decided otherwise so I am not blocking moving forward with for this.

Reply:
What do you think is the right solution?

Reply:
Briefly commented here: https://github.com/hashicorp/design-system/pull/2299#discussion_r1701757266
Essentially, for me the "primary" navigation (the `AppHeader`) should not be blocked/obscured/disabled by the "secondary" navigation (the `SideNav`).
See how the IBM Carbon `AppShell` (not the exact name, but it's what it would be called if we had to implement something similar) works at small viewports. For me this behaviour makes more sense.

Reply:
The overlay/grayscreen of the `SideNav` now no longer overlays the `AppHeader`. I had thought that was your main objection in that old thread which I had resolved since I agreed it probably wasn't the best solution.

I had wanted to keep the `AppHeader`'s child actions still enabled when the `SideNav` was expanded in mobile view, and I explored that solution, but the way the `focus-trap` the `SideNav` is using works I couldn't find a way to do that.

We could look to find another solution in the next major phase of this project but I think it's going to be much more involved (so for me I think the solution of making the actions more visibly look "disabled" is ok for this intermediate phase).

Reply:
OK, I misunderstood then. Better so :)